### ISSUE TITLE
indexer hanging

### ISSUE THREAD

**Issue by atomical** [pat/thinking-sphinx#1055]
Hi Pat,

Yesterday when I ran the indexer it hung on a particular index. Is there a way to debug what is going on? I was originally thinking that it was a locked table, but I'm not so sure now because we cleared the locks and it still wouldn't index.

Adam

**Comment by pat** [comment]
Sphinx's indexer is hanging (as opposed to crashing, or TS code crashing)? I don't think I've seen that before, so I'm really at a loss as to what the cause might be.

Have you tried running the indexer manually, rather than through TS code? Not sure why that'd be different, but worth trying.

**Comment by atomical** [comment]
I just tried running the indexer manually. There are quite a few tables that are open when I run this query in mysql:
```show open tables where in_use>0;```

Is it possible there could be a circular dependency or subquery that is locking one of the required tables for indexing? I'm going to review to see if we've made any changes recently to the indexes, but that's my only guess at this point.

I'm also writing a script to kill the processes that are opening the locks so I can continue to try indexing afresh.
